core/src/it/units/sdm/project/board/gui/GuiBoard.java

```diff
@@ -0,0 +1,141 @@
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.*;
+import com.badlogic.gdx.utils.SnapshotArray;
+import it.units.sdm.project.board.Board;
+import it.units.sdm.project.board.Position;
+import it.units.sdm.project.exceptions.InvalidPositionException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Set;
+import java.util.TreeSet;
+
+public class GuiBoard extends Table implements Board<GuiStone> {
+
+    public static final int TILE_SIZE = 75;
+    public static final Color DARK_TILE = new Color(181 / 255f, 136 / 255f, 99 / 255f, 1);
+    public static final Color LIGHT_TILE = new Color(240 / 255f, 217 / 255f, 181 / 255f, 1);
+    private final int numberOfRows;
+    private final int numberOfColumns;
+    @NotNull
+    private Texture whiteSquareTexture;
+
+    public GuiBoard(@NotNull Skin skin, int numberOfRows, int numberOfColumns) {
+        super(skin);
+        this.numberOfColumns = numberOfColumns;
+        this.numberOfRows = numberOfRows;
+        initTextures();
+        initBoard();
+    }
+
+    private void initTextures() {
+        Pixmap whiteSquare = new Pixmap(TILE_SIZE, TILE_SIZE, Pixmap.Format.RGB565);
+        whiteSquare.setColor(Color.WHITE);
+        whiteSquare.fillRectangle(0, 0, TILE_SIZE, TILE_SIZE);
+        whiteSquareTexture = new Texture(whiteSquare);
+        whiteSquare.dispose();
+    }
+
+    private void initBoard() {
+        for (int i = 0; i < numberOfRows; i++) {
+            row();
+            if (isIndexEven(i)) {
+                initBoardColumns(DARK_TILE, LIGHT_TILE);
+            } else {
+                initBoardColumns(LIGHT_TILE, DARK_TILE);
+            }
+        }
+    }
+
+    private void initBoardColumns(Color oddTilesColor, Color evenTilesColor) {
+        for (int j = 0; j < numberOfColumns; j++) {
+            TextureRegion whiteTextureRegion = new TextureRegion(whiteSquareTexture, 0, 0, TILE_SIZE, TILE_SIZE);
+            Image tile = new Image(whiteTextureRegion);
+            if (isIndexEven(j)) {
+                tile.setColor(evenTilesColor);
+            } else {
+                tile.setColor(oddTilesColor);
+            }
+            Stack tileAndPiece = new Stack(tile);
+            tileAndPiece.addListener(new TileClickListener(tileAndPiece));
+            add(tileAndPiece).size(TILE_SIZE);
+        }
+    }
+
+    private static boolean isIndexEven(int i) {
+        return i % 2 == 0;
+    }
+
+    @Override
+    public boolean isCellOccupied(@NotNull Position position) throws InvalidPositionException {
+        return getPiece(position) != null;
+    }
+
+    @Override
+    public void clearCell(@NotNull Position position) throws InvalidPositionException {
+        for (Cell cell : getCells()) {
+            if (checkCellPosition(position, cell)) {
+                Stack stack = (Stack) cell.getActor();
+                stack.setUserObject(null);
```

Comment:
What is `setUserObject` doing?

Reply:
If the cell is empty, the "UserObject" must be null

Reply:
Never mind I found out on my own, it's about retaining the information on the `Stone`'s color

Reply:
Yep